src/main/java/com/sprint/mission/discodeit/controller/NotificationController.java

```diff
@@ -0,0 +1,67 @@
+package com.sprint.mission.discodeit.controller;
+
+import com.sprint.mission.discodeit.dto.NotificationDto;
+import com.sprint.mission.discodeit.entity.User;
+import com.sprint.mission.discodeit.exception.user.UserNotFoundException;
+import com.sprint.mission.discodeit.repository.UserRepository;
+import com.sprint.mission.discodeit.service.basic.NotificationService;
+import java.nio.file.attribute.UserPrincipal;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Slf4j
+@RestController
+@RequestMapping("/api/notifications")
+@RequiredArgsConstructor
+public class NotificationController {
+
+  private final NotificationService notificationService;
+  private final UserRepository userRepository;
+
+  /**
+   * 알림 조회, 확인(삭제)은 요청자 본인의 알림에 대해서만 수행할 수 있습니다.
+   * <p>
+   * 알림 조회 및 삭제 시 현재 요청의 인증 정보로 요청자를 식별합니다.
+   **/
+  @GetMapping
+  public ResponseEntity<List<NotificationDto>> getNotifications(
+      @AuthenticationPrincipal UserPrincipal userPrincipal
```

Comment:
경린님, 제가 로컬에서 디버깅했을 땐 `UserPrincipal userPrincipal `이 null로 들어와서 에러가 발생하는데요,  

아래와 같이 수정이 필요할 것 같아요!
```suggestion
  @GetMapping
  public ResponseEntity<List<NotificationDto>> getNotifications(
      @AuthenticationPrincipal DiscodeitUserDetails userPrincipal
```